cartography/intel/gsuite/api.py

```diff
@@ -0,0 +1,243 @@
+import logging
+
+from googleapiclient.discovery import HttpError
+
+
+logger = logging.getLogger(__name__)
+
+
+def get_all_groups(admin):
+    """
+    Return list of Google Groups in your organization
+    Returns empty list if we are unable to enumerate the groups for any reasons
+
+    :param admin: apiclient discovery resource object
+    See
+    :return: List of Google groups in domain
+
+    """
+    request = admin.groups().list(customer='my_customer', maxResults=20, orderBy='email')
+    groups = []
+    while request is not None:
+        try:
+            resp = request.execute()
+        except HttpError as e:
+            logger.warning('HttpError occurred in api.get_all_groups(), returning empty list. Details: %r', e)
+            groups = []
+            break
+        groups = groups + resp.get('groups', [])
+        request = admin.groups().list_next(request, resp)
+    return groups
+
+
+def get_all_groups_for_email(admin, email):
+    """ Fetch all groups of which the given group is a member
+
+    Arguments:
+        email: A string representing the email address for the group
+
+    Returns a list of Group models
+    Throws GoogleException
+    """
+    request = admin.groups().list(userKey=email, maxResults=500)
+    groups = []
+    while request is not None:
+        try:
+            resp = request.execute()
+        except HttpError as e:
+            logger.warning('HttpError occurred in api.get_groups_for_email(), returning empty list. Details: %r', e)
+            groups = []
+            break
+        groups = groups + resp.get('groups', [])
+        request = admin.groups().list_next(request, resp)
+    return groups
+
+
+def get_members_for_group(admin, group_email):
+    """ Get all members for a google group
+
+    :param group_email: A string representing the email address for the group
+
+    :return: List of dictionaries representing Users or Groups.
+    """
+    request = admin.members().list(
```

Comment:
Is it possible to pass in the group as a parameter to this function so we don't need to do a `list` twice and possibly get conflicting data?  (maybe a new group was added in the second call)

In https://github.com/lyft/cartography/blob/master/cartography/intel/gcp/compute.py#L939, we get the `zones` in the `sync` function and pass that object as input to `sync_gcp_instances()`.

Reply:
the `list` in this function is actually different than the `list` in the get groups:

`admin.members()` . 
https://developers.google.com/admin-sdk/directory/v1/reference/members/list

vs

`admin.groups()`
https://developers.google.com/admin-sdk/directory/v1/reference/groups/list

Unfortunately you don't get members from the latter and they're unique API calls.